src/App.js

```diff
@@ -1,26 +1,199 @@
+  constructor(){
+    super()
+    this.state = {
+      inputValue: '',
+      tasks: [],
+      activeTasks: [],
+      completedTasks: [],
+      item: 0,
+      clickOnFooter: false
+    }
+
+    this.addTask = this.addTask.bind(this)
+    this.inputTask = this.inputTask.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.removeTask = this.removeTask.bind(this)
+    this.showActive = this.showActive.bind(this)
+    this.showCompleted = this.showCompleted.bind(this)
+    this.showAll = this.showAll.bind(this)
+    this.clearCompleted = this.clearCompleted.bind(this)
+    this.handleChangeFooter = this.handleChangeFooter.bind(this)
+
+
+  }
+
+  inputTask(event) {
+    this.setState({inputValue: event.target.value});
+  }
+
+  addTask(event) {
+    if(event.keyCode === 13){
+      const {inputValue, item} = this.state;
+      this.setState(prevState => {
+        if(inputValue === '')return;
+          const copy = prevState.tasks.concat({})
+          let count = prevState.item + 1
+          copy[item]['text'] = inputValue;
+          copy[item]['checked'] = false;
+          copy[item]['id'] = item
+          return {
+             inputValue:'',
+             tasks: copy,
+             activeTasks: copy,
+             item: count
+           };
+        })
+      }
+    }
+
+  removeTask(index) {
+    this.setState(prevState => {
+      let count = prevState.item;
+      const copy = [...prevState.tasks]
+      if(copy[index].checked === false){
+        count--;
+      }
+       copy.splice(index,1);
+       const updateActiveTask =  this.serchActivTask(copy);
+       const updateCompletedTask = this.updateCompletedTask(copy)
+
+
+
+      return {tasks: copy,
+              item: count,
+              activeTasks: updateActiveTask,
+              completedTasks: updateCompletedTask
+              }
+    })
+
+  }
+
+
+serchActivTask(arr){
+  const filtredArr = arr.filter((el) => {
+      return el.checked === true ? false : true
+      })
+      return filtredArr;
+}
+
+updateCompletedTask(arr){
+  const filtredArr = arr.filter((el) => {
+      return el.checked === true ? true : false
+      })
+      return filtredArr;
+}
+
+
+  handleChange(id) {
+    this.setState(prevState => {
+      let count = prevState.item;
+        const updatedTask = prevState.tasks.map(task => {
+            if (task.id === id) {
+                task.checked === false ? count --: count ++;
+                task.checked = !task.checked
+            }
+            return task
+        })
+
+      const updateActiveTask =  this.serchActivTask(prevState.tasks);
+      const updateCompletedTask = this.updateCompletedTask(prevState.tasks)
+
+        return {
+            tasks: updatedTask,
+            item: count,
+            activeTasks: updateActiveTask,
+            completedTasks: updateCompletedTask
+        }
+      })
+    }
+
+    showActive () {
+      this.setState({tasks: this.state.activeTasks})
```

Comment:
```suggestion
      this.setState({filterBy: 'active'})
```
Then implement method `filterItems` which returns filtered tasks and call it in render:
```javascript
const items = this.filterItems()
...
items.map((item) => ...)
```